src/bridge/BridgeTransport.hpp

```diff
@@ -0,0 +1,178 @@
+/*
+    SlimeVR Code is placed under the MIT license
+    Copyright (c) 2022 SlimeVR Contributors
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in
+    all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+    THE SOFTWARE.
+*/
+#pragma once
+
+#include <filesystem>
+#include <uvw.hpp>
+#include <thread>
+#include <stdint.h>
+
+#include "Logger.hpp"
+#include "CircularBuffer.hpp"
+#include "ProtobufMessages.pb.h"
+
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+    #define LE32_TO_NATIVE(x) (x)
+#else
+    #define LE32_TO_NATIVE(x) ( \
+        ((uint32_t)(x) << 24) | \
+        (((uint32_t)(x) << 8) & 0x00FF0000) | \
+        (((uint32_t)(x) >> 8) & 0x0000FF00) | \
+        ((uint32_t)(x) >> 24) \
+    )
```

Comment:
note: C++23 apparently has std::byteswap in `#include <bits>`. I don't know if we can use that version of the standard yet though. If we don't remove the byteswapping macros like I'm advocating for, we should switch to using the stdlib implementation eventually.